repo/content/content_manager_lock_free.go

```diff
@@ -31,9 +31,17 @@ func (sm *SharedManager) maybeCompressAndEncryptDataForPacking(data gather.Bytes
 
 	// If the content is prefixed (which represents Kopia's own metadata as opposed to user data),
 	// and we're on V2 format or greater, enable internal compression even when not requested.
-	if contentID.HasPrefix() && comp == NoCompression && mp.IndexVersion >= index.Version2 {
-		// 'zstd-fastest' has a good mix of being fast, low memory usage and high compression for JSON.
-		comp = compression.HeaderZstdFastest
+	if contentID.HasPrefix() && mp.IndexVersion >= index.Version2 {
+		mp := sm.format.GetCachedMutableParameters()
+		switch {
+		case mp.MetadataCompression == "none":
+			comp = NoCompression
+		case mp.MetadataCompression != "":
+			comp = compression.ByName[mp.MetadataCompression].HeaderID()
+		default:
+			// For older repo where MetadataCompression is not set, use ZstdFastest algorithm by default
+			comp = compression.ByName[format.DefaultMetadataCompressionAlgorithmName].HeaderID()
+		}
```

Comment:
By deleting `... && comp == NoCompression ` , will we accidentally overwrite `comp` in the `switch` statement that we *passed* to the function? Is it possible if we passed one `comp` value to the function and overwrote it in the `switch` statement by another value? If so, does this have any side effect? In the previous code, the switch run when we passed `NoCompression` only.

If this is what we are expecting, I would then update the comment above the `if` line.

Reply:
@plar switch statement is executed only for content with prefix i.e metadata. As per the existing implementation, for metadata, the comp is always passed as `NoCompression`, and it was hard-coded to set it to `ZstdFastest`. 
With this PR, we want to configure it to the configured value in `MutableParameters`. Regardless of what value is passed (which is always `NoCompression` for metadata), we still want to overwrite it with the configured value as a single source of truth. I will add a comment about it